include/umf/base.h

```diff
@@ -47,7 +47,8 @@ typedef enum umf_result_t {
         6, ///< Failure in user provider code (i.e in user provided callback)
     UMF_RESULT_ERROR_DEPENDENCY_UNAVAILABLE =
         7, ///< External required dependency is unavailable or missing
-    UMF_RESULT_ERROR_UNKNOWN = 0x7ffffffe ///< Unknown or internal error
+    UMF_RESULT_ERROR_INTERNAL = 8,        ///< Internal error
```

Comment:
Can we have something more descriptive than just ERROR_INTERNAL? Perhaps OUT_OF_RESOURCES or even INVALID_NESTING or something like that?

Reply:
`OUT_OF_RESOURCES` is OK for me, 
`INVALID_NESTING` is incomprehensible without knowing the details of the implementation.

Reply:
Done (`OUT_OF_RESOURCES`)